src/btree/bt_cursor.c

```diff
@@ -512,12 +512,35 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
     return (0);
 }
 
+/*
+ * __wt_btcur_reposition --
+ *     Try to evict the page and reposition the cursor on the saved key.
+ */
+int
+__wt_btcur_reposition(WT_CURSOR_BTREE *cbt, bool release_page)
+{
+    WT_CURSOR *cursor;
+    WT_SESSION_IMPL *session;
+
+    cursor = &cbt->iface;
+    session = CUR2S(cbt);
+
+    if (release_page && session->txn->isolation == WT_ISO_SNAPSHOT) { // &&
```

Comment:
Why we need to check release page here? I think it is better to check this at each caller?

Reply:
Updated in [b7c4910](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/b7c4910805456568ee64adda417cacac2776df07)